src/theory/output_channel.h

```diff
@@ -32,6 +32,36 @@
 namespace CVC4 {
 namespace theory {
 
+/** Properties of lemmas */
+enum class LemmaProperty : uint32_t
+{
+  // default
+  NONE = 0,
+  // whether the lemma is removable
+  REMOVABLE = 1,
+  // whether the lemma needs preprocessing
+  PREPROCESS = 2,
+  // whether the processing of the lemma should send atoms to the caller
+  SEND_ATOMS = 4
+};
+/** Define operator lhs | rhs */
+LemmaProperty operator|(LemmaProperty lhs, LemmaProperty rhs);
+/** Define operator lhs |= rhs */
+LemmaProperty& operator|=(LemmaProperty& lhs, LemmaProperty rhs);
+/** Define operator lhs &= rhs */
+LemmaProperty& operator&=(LemmaProperty& lhs, LemmaProperty rhs);
+/** check set lhs & rhs */
+bool operator&(LemmaProperty lhs, LemmaProperty rhs);
```

Comment:
I understand what you do and why you do it, but nevertheless see this as an implementation bug waiting to happen. As it stands, `|` and `&` do significantly different things. For example, I'd expect `lp & (REMOVABLE | SEND_ATOMS)` to return a `LemmaProperty`, not a `bool`. Also `a &= b` does something else than `a = a & b` and `|` and `&` really should be bitwise operators.
I would suggest to have `LemmaProperty operator&(LemmaProperty lhs, LemmaProperty rhs);` and have a separate method `bool abc(LemmaProperty bitmask, LemmaProperty flag)` called `test` or `isSet` or alike.

I guess this is what some people dislike about being able to overload operators in C++ ;-)

```suggestion
LemmaProperty operator&(LemmaProperty lhs, LemmaProperty rhs);
bool test(LemmaProperty lhs, LemmaProperty rhs);
```

Reply:
Yes I totally agree.  And yes, I think I'm convinced now that we should just hardcode methods for testing each bit, e.g.:
```
bool isRemovable(LemmaProperty p);
bool isPreprocess(LemmaProperty p);
bool isSendAtoms(LemmaProperty p);
```